app/src/main/java/co/tinode/tindroid/MessagesFragment.java

```diff
@@ -577,7 +577,9 @@ public void onClick(final View view) {
                             HashMap<String, Object> json = new HashMap<>();
                             json.put("action", "report");
                             json.put("tagret", mTopic.getName());
-                            Cache.getTinode().publish(Tinode.TOPIC_SYS, new Drafty().attachJSON(json));
+                            Drafty msg = new Drafty().attachJSON(json);
+                            Cache.getTinode().publish(Tinode.TOPIC_SYS, msg,
+                                    !msg.isPlain() ? Tinode.getDraftyHeaders(msg) : null);
```

Comment:
The `msg.isPlain()` check is redundant. The earlier call `attachJSON` guarantees that the message is not plain text.

Reply:
Sure.